src/main/java/com/emc/mongoose/metrics/ConcurrentSlidingWindowReservoir.java

```diff
@@ -0,0 +1,52 @@
+package com.emc.mongoose.metrics;
+
+import com.codahale.metrics.Reservoir;
+import com.codahale.metrics.Snapshot;
+import com.codahale.metrics.UniformSnapshot;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+import static java.lang.Math.min;
+
+public class ConcurrentSlidingWindowReservoir
+	implements Reservoir {
+
+	private static final int DEFAULT_SIZE = 1028;
+	private final long[] measurements;
+	private final AtomicLong offset;
+
+	public ConcurrentSlidingWindowReservoir(int size) {
+		this.measurements = new long[size];
+		this.offset = new AtomicLong();
+	}
+
+	public ConcurrentSlidingWindowReservoir() {
```

Comment:
code duplication
should invoke the other constructor with default size as an argument